src/main/java/io/cryostat/net/web/http/api/v1/RecordingUploadPostHandler.java

```diff
@@ -140,9 +135,17 @@ public void handleAuthenticated(RoutingContext ctx) throws Exception {
     }
 
     private ResponseMessage doPost(String recordingName, URL uploadUrl) throws Exception {
-        Path recordingPath =
-                getRecordingPath(recordingName)
-                        .orElseThrow(() -> new HttpStatusException(404, recordingName));
+        Path recordingPath = null;
+        try {
+            recordingPath = recordingArchiveHelper.getRecordingPath(recordingName).get();
+        } catch (ExecutionException e) {
+            Throwable cause = e.getCause();
+            if (cause instanceof RecordingNotFoundException) {
+                throw new HttpStatusException(404, cause.getMessage(), cause);
```

Comment:
I think it would be better to pass `e.getMessage()`/`e` into the `HttpStatusException` constructor rather than `cause`, so we aren't losing a layer of context information, right?